components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -383,4 +394,30 @@ private String resolveImpersonator(JWTClaimsSet claimsSet) {
         }
         return null;
     }
+
+    private void resolveUserStoreDomain(AuthenticatedUser authenticatedUser, String organizationId)
+            throws IdentityOAuth2Exception {
+
+        String userStoreDomain = IdentityUtil.getProperty(ORG_USER_INVITATION_USER_DOMAIN);
```

Comment:
When the userstore is configured as PRIMARY  in this configuration but the userstore mgt configuration level if the admin has renamed the userstore name of PRIMARY, it is wrong to store PRIMARY as the userstore domain.
Let's gracefully handle that case. @DilshanSenarath has done similar fix in another place

Reply:
Better to fix it by introducing a util method to get ORG_USER_INVITATION_USER_DOMAIN resolving this mentioned considtion, instead reading the config directly from IdentityUtil.getProperty(ORG_USER_INVITATION_USER_DOMAIN)

Reply:
@HasiniSama Let's address the comments.

Reply:
Yes. Better to have a util method which encapsulate the logic.